state-chain/traits/src/mocks/broadcaster.rs

```diff
@@ -17,6 +19,49 @@ impl<T> MockPallet for MockBroadcaster<T> {
 	const PREFIX: &'static [u8] = b"MockBroadcaster";
 }
 
+#[derive(
+	Encode,
+	Decode,
+	CloneNoBound,
+	Copy,
+	DefaultNoBound,
+	TypeInfo,
+	PartialEqNoBound,
+	EqNoBound,
+	DebugNoBound,
+	MaxEncodedLen,
+)]
+#[scale_info(skip_type_params(C))]
+pub struct MockApiCall<C> {
+	is_signed: bool,
+	_phantom: PhantomData<C>,
+}
+
+impl<C: ChainCrypto + 'static> ApiCall<C> for MockApiCall<C> {
+	fn threshold_signature_payload(&self) -> <C as cf_chains::ChainCrypto>::Payload {
+		unimplemented!()
+	}
+
+	fn signed(
+		self,
+		_threshold_signature: &<C as cf_chains::ChainCrypto>::ThresholdSignature,
+	) -> Self {
+		Self { is_signed: true, _phantom: Default::default() }
+	}
+
+	fn chain_encoded(&self) -> Vec<u8> {
+		self.encode()
+	}
+
+	fn is_signed(&self) -> bool {
+		self.is_signed
+	}
+
+	fn transaction_out_id(&self) -> <C as cf_chains::ChainCrypto>::TransactionOutId {
+		unimplemented!()
```

Comment:
Is this not leading to a panic during test execution? 🤔

Reply:
Tests are passing so I guess not 😅  ... will take a look.

Reply:
The broadcast pallet mocks use a different MockApiCall (from cf_chains)... 

This Mock is only required to satisfy the trait requirements of the MockBroadcaster... so in fact this `transaction_out_id` never gets called and that's fine.